kv/kvprober: TestPlannerMakesPlansCoveringAllRanges failed

kv/kvprober.TestPlannerMakesPlansCoveringAllRanges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8635615?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8635615?buildTab=artifacts#/) on master @ [dbfc71ba5b5c6bbe16b84d7b807d9bb8f7ba44c0](https://github.com/cockroachdb/cockroach/commits/dbfc71ba5b5c6bbe16b84d7b807d9bb8f7ba44c0):


```
=== RUN   TestPlannerMakesPlansCoveringAllRanges
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges2745556350
    test_log_scope.go:79: use -show-logs to present logs inline
    kvprober_integration_test.go:272: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5358/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:272
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5358/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:296
        	Error:      	Condition never satisfied
        	Test:       	TestPlannerMakesPlansCoveringAllRanges
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges2745556350
--- FAIL: TestPlannerMakesPlansCoveringAllRanges (6.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPlannerMakesPlansCoveringAllRanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24342

kv/kvprober.TestPlannerMakesPlansCoveringAllRanges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8781578?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8781578?buildTab=artifacts#/) on master @ [779c8430ebaddd102a288d4a3dbf118597d97b26](https://github.com/cockroachdb/cockroach/commits/779c8430ebaddd102a288d4a3dbf118597d97b26):


```
=== RUN   TestPlannerMakesPlansCoveringAllRanges
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges4092477712
    test_log_scope.go:79: use -show-logs to present logs inline
    kvprober_integration_test.go:272: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6061/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:272
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6061/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:296
        	Error:      	Condition never satisfied
        	Test:       	TestPlannerMakesPlansCoveringAllRanges
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges4092477712
--- FAIL: TestPlannerMakesPlansCoveringAllRanges (3.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPlannerMakesPlansCoveringAllRanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvprober.TestPlannerMakesPlansCoveringAllRanges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8799745?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8799745?buildTab=artifacts#/) on master @ [b3231c0c80617367fe3f27c3d49c9ae9aad9f95b](https://github.com/cockroachdb/cockroach/commits/b3231c0c80617367fe3f27c3d49c9ae9aad9f95b):


```
=== RUN   TestPlannerMakesPlansCoveringAllRanges
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges1488335924
    test_log_scope.go:79: use -show-logs to present logs inline
    kvprober_integration_test.go:272: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5117/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:272
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5117/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvprober/kvprober_test_/kvprober_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvprober/kvprober_integration_test.go:296
        	Error:      	Condition never satisfied
        	Test:       	TestPlannerMakesPlansCoveringAllRanges
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/172f80d2c7df7011fc0e17bd76629b20/logTestPlannerMakesPlansCoveringAllRanges1488335924
--- FAIL: TestPlannerMakesPlansCoveringAllRanges (4.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPlannerMakesPlansCoveringAllRanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvprober/kvprober_integration_test.go
+++ b/pkg/kv/kvprober/kvprober_integration_test.go
@@ -256,7 +256,10 @@ func TestPlannerMakesPlansCoveringAllRanges(t *testing.T) {
 	skip.UnderShort(t)
 
 	ctx := context.Background()
-	_, sqlDB, p, cleanup := initTestProber(t, base.TestingKnobs{})
+	// Disable split and merge queue just in case.
+	_, sqlDB, p, cleanup := initTestProber(t, base.TestingKnobs{
+		Store: &kvserver.StoreTestingKnobs{DisableSplitQueue: true, DisableMergeQueue: true},
+	})
 	defer cleanup()
 
 	rangeIDToTimesWouldBeProbed := make(map[int64]int)
@@ -290,7 +293,7 @@ func TestPlannerMakesPlansCoveringAllRanges(t *testing.T) {
 				}
 			}
 			return true
-		}, time.Second, time.Millisecond)
+		}, testutils.DefaultSucceedsSoonDuration, 20*time.Millisecond)
 	}
 	for i := 0; i < 20; i++ {
 		test(i)

